fuzzing/private/binary.bzl

```diff
@@ -14,7 +14,7 @@
 
 """Defines a rule for creating an instrumented fuzzing executable."""
 
-load("//fuzzing/private:engine.bzl", "CcFuzzingEngineInfo")
+load("//fuzzing/private:engine.bzl", "FuzzingEngineInfo")
```

Comment:
The `CcFuzzing*` -> `Fuzzing*` refactoring seems to impact quite a few files. Should we do it as a separate PR, to facilitate reviewing and keeping the changes more focused?

Reply:
Yes, I'll split this up.

Reply:
Maybe let's hold this off for a bit until we make a decision on the provider structure below.